decorator.rb

```diff
@@ -0,0 +1,12 @@
+require './nameable'
+
+class Decorator < Nameable
+  def initialize(nameable)
+    @decoration = nameable
```

Comment:
Kindly set the getters and setters for the `@decoration` instance variable. You can do so by setting `attr_accessor` for the same.